catalog/cataloger_diff.go

```diff
@@ -215,7 +245,28 @@ func (c *cataloger) diffFromChild(tx db.Tx, childID, parentID int64) error {
 	return nil
 }
 
-func (c *cataloger) diffNonDirect(_ db.Tx, leftID, rightID int64) error {
+// contextWithDiffResultsDispose generate diff results id used for temporary table name
+func contextWithDiffResultsDispose(ctx context.Context, tx sq.Execer) (context.Context, context.CancelFunc) {
+	id := xid.New().String()
+	return context.WithValue(ctx, contextDiffResultsKey, id), func() {
+		tableName := diffResultsTableNameFormat(id)
+		_, _ = tx.Exec("DROP TABLE IF EXISTS " + tableName)
+	}
+}
+
+func diffResultsTableNameFromContext(ctx context.Context) (string, error) {
+	id, ok := ctx.Value(contextDiffResultsKey).(string)
+	if !ok {
+		return "", ErrMissingDiffResultsIDInContext
+	}
+	return diffResultsTableNameFormat(id), nil
+}
+
+func diffResultsTableNameFormat(id string) string {
+	return diffResultsTableNamePrefix + "_" + id
```

Comment:
Can we use `fmt.Sprintf` here?

Reply:
Prefer not unless it is not clear.

1. Think is readable - I can get the two constants together if it will be cleaner
2. I use Sprintf when needed - 2 strings concatenation using Sprintf is not the case
3. Performance
 
```
goos: darwin
goarch: amd64
pkg: bench
BenchmarkConcat-16      10384356               113 ns/op
BenchmarkSprintf-16      5039378               241 ns/op
PASS
ok      bench   3.290s
```